### ISSUE TITLE
Flaky-test: BrokerInterceptorTest.testProducerCreation

### ISSUE THREAD

**Issue by codelipenghui** [apache/pulsar#16516]
[example failure](https://github.com/apache/pulsar/runs/7274700601?check_suite_focus=true)

```
Error:  Tests run: 18, Failures: 3, Errors: 0, Skipped: 9, Time elapsed: 128.822 s <<< FAILURE! - in org.apache.pulsar.broker.intercept.BrokerInterceptorTest
Error:  testProducerCreation(org.apache.pulsar.broker.intercept.BrokerInterceptorTest)  Time elapsed: 0.247 s  <<< FAILURE!
java.lang.AssertionError: expected [true] but found [false]
	at org.testng.Assert.fail(Assert.java:99)
	at org.testng.Assert.failNotEquals(Assert.java:1037)
	at org.testng.Assert.assertTrue(Assert.java:45)
	at org.testng.Assert.assertTrue(Assert.java:55)
	at org.apache.pulsar.broker.intercept.BrokerInterceptorTest.testProducerCreation(BrokerInterceptorTest.java:145)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:132)
	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:45)
	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:73)
	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:11)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

<!-- optionally provide the full stacktrace ->
<details>
<summary>Full exception stacktrace</summary>
<code><pre>
full exception stacktrace here
</pre></code>
</details>

**Comment by Technoboy-** [comment]
Closed by #16742

**Comment by codelipenghui** [comment]
Another one: https://github.com/apache/pulsar/runs/7850960846?check_suite_focus=true
